### PR TITLE
add trx service

### PR DESCRIPTION
**Pull request by ttaoS**

### CODE REVIEW

**Comment by ttaoS** [comment]
> signAndSend

Cool. I see. We do have API context in store, but we seem don't populate API when connecting api. Account balance service seems also create new API each time. We should also make account balance reuse the API if it can, right?

image

File: src/ts/services/transactions/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  network: Network): Promise<string> => {
+
+  const provider = new WsProvider(network.rpcServer)
+
+  // Instantiate the API
+  const api = await ApiPromise.create(provider)
+
+  const keyring = new Keyring({ type: 'sr25519' })
+
+  const from = keyring.addFromJson(keyringPair$Json)
```
Comment (hskang9) [inline src/ts/services/transactions/index.ts]: if we are going to use `addFromJson`, I think we would need to check if the key is locked and ask for password to unlock it.

Comment (ttaoS) [reply src/ts/services/transactions/index.ts]: yeah. good point

Comment (hskang9) [reply src/ts/services/transactions/index.ts]: you can add logic:
```node
if(keyring.islocked()) {
// ask for pw
}

**Comment by fyang1024** [comment]
> > signAndSend
> 
> Cool. I see. We do have API context in store, but we seem don't populate API when connecting api. Account balance service seems also create new API each time. We should also make account balance reuse the API if it can, right?
> 
> image

Balance component does reuse the api in IAppContext.
We populate the API asynchronously during PopupApp Initialization, as we cannot set a value for api at the beginning, that's why it's declared optional in IAppContext. apiReady is used as a handy flag to check if api is set in IAppContext.

File: src/ts/services/transactions/index.ts (hunk 1 above)
Comment (ttaoS) [reply src/ts/services/transactions/index.ts]: I will probably move the transaction signing logic in keyring-vault. I will push a change later to demonstrate it.

**Comment by ttaoS** [comment]
close it for now. will reopen when it is ready

**Pull request closed without merging**